packages/perseus/src/widgets/categorizer/validate-categorizer.ts

```diff
@@ -0,0 +1,28 @@
+import type {PerseusStrings} from "../../strings";
+import type {PerseusScore} from "../../types";
+import type {
+    PerseusCategorizerRubric,
+    PerseusCategorizerUserInput,
+} from "../../validation.types";
+
+function validateCategorizer(
+    userInput: PerseusCategorizerUserInput,
+    rubric: PerseusCategorizerRubric,
```

Comment:
CC @jeremywiebe 

I think this highlights a possible need for more even more clarification of types (or maybe just a helper):
- [Example using unions (seems cleaner)](https://www.typescriptlang.org/play/?#code/C4TwDgpgBASgrgIwE4EsDGUC8UDeAoKQqFYCAWwGcAuKAOzjIQiQG0BdAbjwF889RIUAIK0KAd2ZZcBImgD2SJBDTAa9Rs3Zde-cNHjJ0AdRIALEeMnYDqDADJhoiUi555o4FACGT5jQvOUvhEUPKKyqpQLAAcADRQAGzxAOzxAKzxAMzxAAzxAJxsPHzuFJ4AZhA26DTVGNjBRCTk1FFxiSnpWbkFRTqlnkx1tYi2JsDmvkhBMoTNlDQx8UlQqVAZUNlQeVCFsbOhCkoqi+0raxtbO4U8QA)
- [Example using utilities](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAIglgMwRAThAdgYwgHgCoA0UAqgHxQC8UACnJgNb5ECiAHpgDYCuAJrvRBAB7BFEJQBw0WVIBuAFDzQkKACUuAIxR1KUAN7yoRqHGAQAtgGcAXFHRdzG1AG0AuguNRMQlGkzBbe0cXd3kAX0VlaABBdEsAd1RdWgYcdS06IgAib18Ifyy5SPBodO1MAHVTAAshLmBYhKSqeCRUDGw0zXKiRsSUIvlvOOAoAEM4-ts+5v1DY1y-AKhnAA4iADYiAHYiAFYiAGYiAAYiAE5XcMVhy1HkMrpbR8qauobJ2YNPUwsbFfWUC2UF2UAOUGOUDOUEu1yGQhGUCcL2e3R0VG+xl+Vlsa02O32R1OF1cBHmRkW+WWeKBBLBRKhJPCQA)